### PR TITLE
SAAS-24870 adding open-jdk17 to java-releases

### PR DESCRIPTION
**Pull request by niharbabu99**

### CODE REVIEW

File: ci/pipeline.yml [hunk 1]
```diff
@@ -43,6 +43,30 @@ jobs:
       rebase: true
       repository: java-release-out
 
+- name: build-11
```
Comment (nimeshjm) [inline ci/pipeline.yml]: build-17?

File: ci/pipeline.yml [hunk 2]
```diff
@@ -43,6 +43,30 @@ jobs:
...
+  - in_parallel:
+    - get: daily
+      trigger: true
+    - get: java-release
+    - get: stemcell
+      trigger: true
+  - task: get_release
+    file: java-release/ci/get_release.yml
+    params:
+      JAVA_VERSION: 11
```
Comment (nimeshjm) [inline ci/pipeline.yml]: 17?

File: ci/pipeline.yml (hunk 1 above)
Comment (niharbabu99) [reply ci/pipeline.yml]: Realized that while validating locally and rectified it.

**Comment by niharbabu99** [comment]
```
-----> Wed Oct 12 12:44:50 UTC 2022: Run test errand
Using environment '10.245.0.3' as client 'admin'

Using deployment 'test'

Task 4

Task 4 | 12:44:51 | Preparing deployment: Preparing deployment
Task 4 | 12:44:51 | Preparing package compilation: Finding packages to compile (00:00:00)
Task 4 | 12:44:51 | Preparing deployment: Preparing deployment (00:00:00)
Task 4 | 12:44:51 | Creating missing vms: openjdk-test/c7694c05-b840-4cab-92f0-117ea589e8d2 (0) (00:00:05)
Task 4 | 12:44:56 | Updating instance openjdk-test: openjdk-test/c7694c05-b840-4cab-92f0-117ea589e8d2 (0) (canary)
Task 4 | 12:44:59 | L executing pre-stop: openjdk-test/c7694c05-b840-4cab-92f0-117ea589e8d2 (0) (canary)
Task 4 | 12:44:59 | L executing drain: openjdk-test/c7694c05-b840-4cab-92f0-117ea589e8d2 (0) (canary)
Task 4 | 12:45:00 | L stopping jobs: openjdk-test/c7694c05-b840-4cab-92f0-117ea589e8d2 (0) (canary)
Task 4 | 12:45:01 | L executing post-stop: openjdk-test/c7694c05-b840-4cab-92f0-117ea589e8d2 (0) (canary)
Task 4 | 12:45:02 | L installing packages: openjdk-test/c7694c05-b840-4cab-92f0-117ea589e8d2 (0) (canary)
Task 4 | 12:45:03 | L configuring jobs: openjdk-test/c7694c05-b840-4cab-92f0-117ea589e8d2 (0) (canary)
Task 4 | 12:45:03 | L executing pre-start: openjdk-test/c7694c05-b840-4cab-92f0-117ea589e8d2 (0) (canary)
Task 4 | 12:45:03 | L starting jobs: openjdk-test/c7694c05-b840-4cab-92f0-117ea589e8d2 (0) (canary)
Task 4 | 12:45:09 | L executing post-start: openjdk-test/c7694c05-b840-4cab-92f0-117ea589e8d2 (0) (canary) (00:00:13)
Task 4 | 12:45:09 | Running errand: openjdk-test/c7694c05-b840-4cab-92f0-117ea589e8d2 (0) (00:00:01)
Task 4 | 12:45:10 | Fetching logs for openjdk-test/c7694c05-b840-4cab-92f0-117ea589e8d2 (0): Finding and packing log files (00:00:01)

Task 4 Started  Wed Oct 12 12:44:51 UTC 2022
Task 4 Finished Wed Oct 12 12:45:11 UTC 2022
Task 4 Duration 00:00:20
Task 4 done

Instance   openjdk-test/c7694c05-b840-4cab-92f0-117ea589e8d2  
Exit Code  0  
Stdout     javac 17.0.4.1  
             
Stderr     + source /var/vcap/packages/openjdk-17/bosh/compile.env  
           ++ export JAVA_HOME=/var/vcap/packages/openjdk-17  
           ++ JAVA_HOME=/var/vcap/packages/openjdk-17  
           ++ export PATH=/var/vcap/packages/openjdk-17/bin:/usr/sbin:/usr/bin:/sbin:/bin  
           ++ PATH=/var/vcap/packages/openjdk-17/bin:/usr/sbin:/usr/bin:/sbin:/bin  
           + javac -version  
           + source /var/vcap/packages/openjdk-17/bosh/runtime.env  
           ++ export JAVA_HOME=/var/vcap/packages/openjdk-17  
           ++ JAVA_HOME=/var/vcap/packages/openjdk-17  
           ++ export PATH=/var/vcap/packages/openjdk-17/bin:/usr/sbin:/usr/bin:/sbin:/bin  
           ++ PATH=/var/vcap/packages/openjdk-17/bin:/usr/sbin:/usr/bin:/sbin:/bin  
           + java -version  
           openjdk version "17.0.4.1" 2022-08-12  
           OpenJDK Runtime Environment Temurin-17.0.4.1+1 (build 17.0.4.1+1)  
           OpenJDK 64-Bit Server VM Temurin-17.0.4.1+1 (build 17.0.4.1+1, mixed mode, sharing)  
             

1 errand(s)

Succeeded
```

Running a test errand is successful in my local setup.

**Review by aramprice — COMMENTED** [review]
The final changes look good. The four commits should be squashed into one.

Thought: is the manifest change you made, and then reverted an indication that the CI for this repo is missing another dimension of testing?

Phrased another way:
> Should this release be tested across a set of different stemcells?

**Comment by niharbabu99** [comment]
> The final changes look good. The four commits should be squashed into one.
> 
> Thought: is the manifest change you made, and then reverted an indication that the CI for this repo is missing another dimension of testing?
> 
> Phrased another way:
> 
> > Should this release be tested across a set of different stemcells?

@aramprice Thanks for taking a look into PR.
Ideally, JDK should work on both xenial and bionic, as Our system son bionic I just want to be double sure about it.

Regarding the CI for this repo, it can be enhanced with the below ones
1. Pipeline setting itself based on the commit to pipeline config. 
2. There are a lot of duplicate lines of code, which can be removed.
3. We can enhance to test on both xenial and bionic and have the flexibility to include new ones in the future when they appear.
If required I can create a separate PR for the above ones once this is merged.

Let me try to squash the 4 commits and re-submit the PR.

**Comment by aramprice** [comment]
Thanks for taking a look at squashing the commits, I think doing a `git push --force-with-lease` should update this PR, without needing to create a new one.

Thank you for the background info on the state of CI. For this PR it probably doesn't make sense to address the CI difficulties.

**Comment by niharbabu99** [comment]
@aramprice I have squashed all 4 commits to 1.

**Comment by niharbabu99** [comment]
> Thanks for taking a look at squashing the commits, I think doing a `git push --force-with-lease` should update this PR, without needing to create a new one.
 I meant a separate PR for CI enhancements, not for the squash :)

**Review by aramprice — APPROVED** [review]
Looks good, thank you! 

I think it would be good to update the commit message to remove the squashed comments. That will be less confusing for folks in the future.

**Comment by niharbabu99** [comment]
> Looks good, thank you!
> 
> I think it would be good to update the commit message to remove the squashed comments. That will be less confusing for folks in the future.

Done.

**Review by nimeshjm — APPROVED** [review]

**Comment by niharbabu99** [comment]
@aramprice can we get this merged? 

Once merged who sets the pipeline so that the new pipeline configuration is applied?

**Comment by aramprice** [comment]
@niharbabu99 I'm unsure about the process for updating pipelines. Tagging @rkoster who likely knows if there is a particular process for that.

As for merging, I'm less familiar with this area of the BOSH ecosystem so I'd like to have @rkoster (or someone he designates) do the final merge. This might happen on Monday though because today is a company holiday for VMware.

**Comment by niharbabu99** [comment]
@rkoster can we get this PR merged and set the new CI pipeline config to get JDK17 available

**Comment by rkoster** [comment]
Pipeline has been migrated to Cloudfoundry Foundation owned concourse: https://github.com/bosh-packages/java-release/pull/17

**Review by rkoster — APPROVED** [review]

**Pull request merged**